components/script/dom/mutationobserver.rs

```diff
@@ -135,6 +136,26 @@ impl MutationObserver {
                         }
                         // Step 3.1.2
                         let paired_string = if registered.options.attribute_old_value {
+                            old_value.clone()
+                        } else {
+                            None
+                        };
+                        // Step 3.1.1
+                        let idx = interestedObservers.iter().position(|&(ref o, _)|
+                            &**o as *const _ == &*registered.observer as *const _);
+                        if let Some(idx) = idx {
+                            interestedObservers[idx].1 = paired_string;
```

Comment:
nit: You want `find` instead of `position` + the index.

Reply:
Using find() leads to a bunch of problems with needing to borrow mutably interested_observer twice, and working around that is actually more code than currently and hurts readability quite a lot.
So unless I'm missing a trick, I'd rather keep it like that.